### PR TITLE
Minor improvements to EIP-7685

### PR DESCRIPTION
**Pull request by Gabriel-Trintinalia**
## PR description
Address feedback of #7068

### CODE REVIEW

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/requests/RequestUtil.java [hunk 1]
```diff
@@ -40,6 +42,23 @@ public static <T extends Request> List<T> filterRequestsOfType(
...
+
+  public static Optional<List<WithdrawalRequest>> getWithdrawalRequests(
+      final Optional<List<Request>> requests) {
+    return requests.map(r -> filterRequestsOfType(r, WithdrawalRequest.class));
+  }
+
+  /**
+   * Combines two optional lists of requests into a single optional list.
+   *
+   * @param maybeDeposits Optional list of deposit requests.
```
Comment (macfarla) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/requests/RequestUtil.java:57]: nit - for consistency should this be renamed to maybeDepositsRequest

Comment (macfarla) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/requests/RequestUtil.java:57]: actually on further consideration I see that matches the class names Deposit and WithdrawalRequest

Comment (macfarla) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/requests/RequestUtil.java:57]: We have Withdrawal and WithdrawalRequest but Deposit is itself the request... it is a bit confusing but probably outside the scope of this PR. non-blocking.

**Review by macfarla — APPROVED** [review]

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/requests/RequestUtil.java (hunk 1 above)
Comment (Gabriel-Trintinalia) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/requests/RequestUtil.java:57]: I will rename Deposit to DepositRequest but I wanted to do that in a different PR so it is easier to review it.

**Review by jframe — APPROVED** [review]

**Pull request merged**